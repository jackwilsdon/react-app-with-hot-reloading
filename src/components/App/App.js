import PropTypes from 'prop-types';
import React, { Component } from 'react';
import logo from '../../assets/logo.svg';
import './App.css';
import Counter from '../Counter';
import { Provider } from 'react-redux';

class App extends Component {
  static propTypes = {
    store: PropTypes.any.isRequired,
  };

  render() {
    return (
      <Provider store={this.props.store}>
        <div className="App">
          <header className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <h1 className="App-title">Welcome to React</h1>
          </header>
          <div className="App-intro">
            <Counter />
          </div>
        </div>
      </Provider>
    );
  }
}

export default App;
