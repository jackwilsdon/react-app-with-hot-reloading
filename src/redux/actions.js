import * as actions from '../actions';
import pickBy from 'lodash/pickBy';
import isFunction from 'lodash/isFunction';

export default pickBy(actions, action => isFunction(action));
