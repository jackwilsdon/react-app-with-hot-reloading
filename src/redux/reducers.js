import * as reducers from '../reducers';

export { reducers as default };
