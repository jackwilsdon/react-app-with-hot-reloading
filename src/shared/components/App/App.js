import './App.css';
import React, { Component } from 'react';
import Counter from '../Counter';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import logo from '../../assets/logo.svg';

class App extends Component {
  static propTypes = {
    store: PropTypes.any.isRequired,
  };

  render() {
    return (
      <Provider store={this.props.store}>
        <div className="App">
          <header className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <h1 className="App-title">Welcome to React</h1>
          </header>
          <div className="App-intro">
            <Counter />
          </div>
        </div>
      </Provider>
    );
  }
}

export default App;
