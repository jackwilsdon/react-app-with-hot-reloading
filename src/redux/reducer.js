import { combineReducers } from 'redux';
import counter from '../counter/ducks';

export default combineReducers({ counter });
