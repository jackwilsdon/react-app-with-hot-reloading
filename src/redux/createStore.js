import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import actions from './actions';
import { connectRoutes } from 'redux-first-router';
import createBrowserHistory from 'history/createBrowserHistory';
import reducers from './reducers';
import routes from '../router/routes';

const composeEnhancers =
  process.env.NODE_ENV === 'production' ||
  !window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? compose
    : window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        actionCreators: actions,
      });

export default (initialState = {}) => {
  const history = createBrowserHistory();
  const { reducer, middleware, enhancer } = connectRoutes(history, routes);

  const rootReducer = combineReducers({
    ...reducers,
    location: reducer,
  });

  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancers(enhancer, applyMiddleware(middleware)),
  );

  // Support reloading reducers if hot reloading is enabled.
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      const reducers = require('./reducers').default;

      const rootReducer = combineReducers({
        ...reducers,
        location: reducer,
      });

      store.replaceReducer(rootReducer);
    });
  }

  return store;
};
