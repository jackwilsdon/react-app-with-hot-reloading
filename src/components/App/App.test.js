import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { createStore } from 'redux';
import identity from 'lodash/identity';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const store = createStore(identity, {});
  const tree = renderer.create(<App store={store} />).toJSON();

  expect(tree).toMatchSnapshot();
});
