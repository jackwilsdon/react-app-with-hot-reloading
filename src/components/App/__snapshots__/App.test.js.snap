// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <div
    className="App-intro"
  >
    <div
      className="Counter"
    >
      <button
        className="Counter__button"
        onClick={[Function]}
      >
        -
      </button>
      <div
        className="Counter__value"
      />
      <button
        className="Counter__button"
        onClick={[Function]}
      >
        +
      </button>
    </div>
  </div>
</div>
`;
